### PR TITLE
[Spring] Add new params for ACS settings

### PR DESCRIPTION
**Pull request by moarychan**
---

Add new params for ACS.

### Related command
- Add argument `--refresh-interval` in `spring application-configuration-service create`
- Add argument `--refresh-interval` in `spring application-configuration-service update`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

File: src/spring/azext_spring/_params.py [hunk 1]
```diff
@@ -864,6 +865,11 @@ def prepare_logs_argument(c):
     for scope in ['create', 'update']:
         with self.argument_context('spring application-configuration-service {}'.format(scope)) as c:
             c.argument('generation', arg_type=get_enum_type(ConfigurationServiceGeneration), help='Generation of Application Configuration Service.')
+            c.argument('refresh_interval', type=int,
+                       validator=validate_refresh_interval,
+                       help='Number of the repository refresh interval (in seconds). '
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:870]: 'Specify the interval (in seconds) for refreshing the repository. Use 0 to turn off automatic refresh. An interval of at least 60 seconds is recommended.'

File: src/spring/azext_spring/_validators_enterprise.py [hunk 2]
```diff
@@ -352,6 +352,12 @@ def validate_acs_create(namespace):
             raise ArgumentUsageError("--application-configuration-service-generation can only be set when enable application configuration service.")
 
 
+def validate_refresh_interval(namespace):
+    if namespace.refresh_interval is not None:
+        if namespace.refresh_interval < 0:
```
Comment (yuwzho) [inline src/spring/azext_spring/_validators_enterprise.py:357]:
```suggestion
    if namespace.refresh_interval and namespace.refresh_interval < 0:
```

File: src/spring/azext_spring/application_configuration_service.py [hunk 3]
```diff
@@ -24,25 +24,35 @@
...
+                                             generation=None, refresh_interval=None):
     if generation is None:
         generation = ConfigurationServiceGeneration.GEN1
 
-    properties = models.ConfigurationServiceProperties(generation=generation)
+    if refresh_interval is not None and refresh_interval >= 0:
+        properties = models.ConfigurationServiceProperties(generation=generation,
+                                                           settings={"refresh_interval_in_seconds": refresh_interval})
+    else:
+        properties = models.ConfigurationServiceProperties(generation=generation)
```
Comment (yuwzho) [inline src/spring/azext_spring/application_configuration_service.py:35]:
```suggestion
   properties = models.ConfigurationServiceProperties(generation=generation)
    if refresh_interval is not None:
        properties.settings = models.ConfigurationServiceSettings(refresh_interval_in_seconds=refresh_interval)
```

File: src/spring/azext_spring/application_configuration_service.py [hunk 4]
```diff
@@ -24,25 +24,35 @@
...
         acs_resource.properties.generation = generation
         logger.warning("Updating with generation {}".format(generation))
     else:
         acs_resource.properties.generation = ConfigurationServiceGeneration.GEN1
         logger.warning("Default generation will be Gen1")
+    if refresh_interval is not None and refresh_interval >= 0:
+        if acs_resource.properties.settings is None:
+            acs_resource.properties.settings = {"refresh_interval_in_seconds": refresh_interval}
+        else:
+            acs_resource.properties.settings.refresh_interval_in_seconds = refresh_interval
```
Comment (yuwzho) [inline src/spring/azext_spring/application_configuration_service.py:54]:
```suggestion
    if refresh_interval is not None:
       acs_resource.properties.settings =  acs_resource.properties.settings or models.ConfigurationServiceSettings()
       acs_resource.properties.settings.refresh_interval_in_seconds = refresh_interval
```

File: src/spring/azext_spring/tests/latest/test_asa_application_configuration_service.py [hunk 5]
```diff
@@ -91,5 +91,17 @@ def test_application_configuration_service(self, resource_group, spring, app):
             self.check('properties.provisioningState', "Succeeded")
         ])
 
-        self.cmd('spring application-configuration-service update -g {rg} -s {serviceName} --generation Gen2',
-                 checks=[self.check('properties.provisioningState', "Succeeded")])
+        self.cmd('spring application-configuration-service update -g {rg} -s {serviceName} '
```
Comment (yuwzho) [inline src/spring/azext_spring/tests/latest/test_asa_application_configuration_service.py:94]: Please add some UnitTest to validate the command code. You can refer the `TestSpringCloudCreateEnterprise` inside the `test_asa_create.py` file.

**Review by yuwzho — APPROVED** [review]
LGTM

**Comment by moarychan** [comment]
Hi @zhoxing-ms @yanzhudd @FumingZhang, could you help to merge this PR? Many thanks!

**Comment by moarychan** [comment]
Hi @yonzhan, could you help to merge this PR? Thanks！

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**